### PR TITLE
Ignite 15931 2

### PR DESCRIPTION
**Pull request by denis-chudov**

### CODE REVIEW

File: modules/core/src/main/java/org/apache/ignite/hlc/HybridTimestamp.java [hunk 1]
```diff
@@ -17,13 +17,14 @@
...
 
+import java.io.Serializable;
 import org.apache.ignite.internal.tostring.S;
 import org.jetbrains.annotations.Nullable;
 
 /**
  * A hybrid timestamp that combines physical clock and logical clock.
  */
-public class HybridTimestamp implements Comparable<HybridTimestamp> {
+public class HybridTimestamp implements Comparable<HybridTimestamp>, Serializable {
```
Comment (sanpwc) [inline modules/core/src/main/java/org/apache/ignite/hlc/HybridTimestamp.java:27]: Why? @Marshallable is enough for proper message fields serialization.

File: modules/transactions/src/main/java/org/apache/ignite/internal/tx/storage/state/TxStateStorage.java [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+     * @return Future which completes when the snapshot operation is complete.
+     */
+    CompletableFuture<Void> snapshot(Path snapshotPath);
+
+    /**
+     * Restore a storage from the snapshot.
+     *
+     * @param snapshotPath Snapshot path.
+     */
+    void restoreSnapshot(Path snapshotPath);
```
Comment (sanpwc) [inline modules/transactions/src/main/java/org/apache/ignite/internal/tx/storage/state/TxStateStorage.java:115]: We will also need scan in order to retrieve all existing tx meta, e.g. on primary replica switch including node restart.

File: modules/core/src/main/java/org/apache/ignite/hlc/HybridTimestamp.java (hunk 1 above)
Comment (denis-chudov) [reply modules/core/src/main/java/org/apache/ignite/hlc/HybridTimestamp.java:27]: It's necessary for `ByteUtils#toBytes`.

File: modules/transactions/src/main/java/org/apache/ignite/internal/tx/storage/state/TxStateStorage.java (hunk 2 above)
Comment (denis-chudov) [reply modules/transactions/src/main/java/org/apache/ignite/internal/tx/storage/state/TxStateStorage.java:115]: added `scan`.

File: modules/core/src/main/java/org/apache/ignite/hlc/HybridTimestamp.java (hunk 1 above)
Comment (sergeyuttsel) [inline modules/core/src/main/java/org/apache/ignite/hlc/HybridTimestamp.java:27]: serialVersionUID is missed.

File: modules/transactions/src/main/java/org/apache/ignite/internal/tx/TxMeta.java [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+package org.apache.ignite.internal.tx;
+
+import static java.util.Collections.unmodifiableList;
+
+import java.io.Serializable;
+import java.util.List;
+import org.apache.ignite.hlc.HybridTimestamp;
+
+/** Transaction meta. */
+public class TxMeta implements Serializable {
```
Comment (sergeyuttsel) [inline modules/transactions/src/main/java/org/apache/ignite/internal/tx/TxMeta.java:27]: serialVersionUID is missed.

**Pull request closed without merging**